pkg/netutil/netutil.go

```diff
@@ -171,76 +246,75 @@ func (e *CNIEnv) networkConfigList() ([]*networkConfig, error) {
 				return nil, err
 			}
 		}
+		id, labels := nerdctlIDLabels(lcl.Bytes)
 		l = append(l, &networkConfig{
 			NetworkConfigList: lcl,
-			NerdctlID:         nerdctlID(lcl.Bytes),
-			NerdctlLabels:     nerdctlLabels(lcl.Bytes),
+			NerdctlID:         id,
+			NerdctlLabels:     labels,
 			File:              fileName,
 		})
 	}
 	return l, nil
 }
 
-// AcquireNextID suggests the next ID.
-func (e *CNIEnv) AcquireNextID() (int, error) {
-	maxID := DefaultID
-	for _, n := range e.Networks {
-		if n.NerdctlID != nil && *n.NerdctlID > maxID {
-			maxID = *n.NerdctlID
-		}
-	}
-	nextID := maxID + 1
-	return nextID, nil
-}
-
-func nerdctlID(b []byte) *int {
-	type nerdctlConfigList struct {
-		NerdctlID *int `json:"nerdctlID,omitempty"`
+func nerdctlIDLabels(b []byte) (*string, *map[string]string) {
+	type idLabels struct {
+		ID     *string            `json:"nerdctlID,omitempty"`
+		Labels *map[string]string `json:"nerdctlLabels,omitempty"`
 	}
-	var ncl nerdctlConfigList
-	if err := json.Unmarshal(b, &ncl); err != nil {
-		// The network is managed outside nerdctl
-		return nil
-	}
-	return ncl.NerdctlID
-}
-
-func nerdctlLabels(b []byte) *map[string]string {
-	type nerdctlConfigList struct {
-		NerdctlLabels *map[string]string `json:"nerdctlLabels,omitempty"`
-	}
-	var ncl nerdctlConfigList
-	if err := json.Unmarshal(b, &ncl); err != nil {
-		return nil
+	var idl idLabels
+	if err := json.Unmarshal(b, &idl); err != nil {
+		return nil, nil
 	}
-	return ncl.NerdctlLabels
+	return idl.ID, idl.Labels
 }
 
-func GetBridgeName(id int) string {
-	return fmt.Sprintf("nerdctl%d", id)
+func networkID(name string) string {
+	hash := sha256.Sum256([]byte(name))
+	return hex.EncodeToString(hash[:])
 }
 
-func parseIPAMRange(subnetStr, gatewayStr, ipRangeStr string) (*IPAMRange, error) {
-	subnetIP, subnet, err := net.ParseCIDR(subnetStr)
+func (e *CNIEnv) parseSubnet(subnetStr string) (*net.IPNet, error) {
+	usedSubnets, err := e.usedSubnets()
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse subnet %q", subnetStr)
+		return nil, err
 	}
-	if !subnet.IP.Equal(subnetIP) {
-		return nil, fmt.Errorf("unexpected subnet %q, maybe you meant %q?", subnetStr, subnet.String())
+	var subnet *net.IPNet
+	if subnetStr == "" {
```

Comment:
Suggest: this branch can be moved to the beginning of function and fast return instead of using else branch~

Reply:
I don't get it. Can you get me more explain?

Reply:
I got it.